ga4gh/backend.py

```diff
@@ -217,6 +219,62 @@ def variantSetsGenerator(self, request):
         return self._topLevelObjectGenerator(
             request, self._variantSetIdMap, self._variantSetIds)
 
+    def readsGenerator(self, request):
+        # Local utility function to save some typing
+        def getPosition(readAlignment):
+            return readAlignment.alignment.position.position
+
+        if len(request.readGroupIds) != 1:
+            raise exceptions.NotImplementedException(
+                "Read search over multiple readGroups not supported")
+        readGroupId = request.readGroupIds[0]
+        try:
+            readGroup = self._readGroupIdMap[request.readGroupIds[0]]
+        except KeyError:
+            raise exceptions.ReadGroupNotFoundException(readGroupId)
+        startPosition = request.start
+        equalPositionsToSkip = 0
+        if request.pageToken is not None:
+            startPosition, equalPositionsToSkip = self.parsePageToken(
+                request.pageToken, 2)
```

Comment:
Testing should definitely not be done through a simulator (either the end to end one we have implemented or otherwise).  We should just create a discrete set of cases that exercise every branch.  But that's for another PR.